homeassistant/components/climate/toon.py

```diff
@@ -0,0 +1,87 @@
+"""
+Toon van Eneco Thermostat Support.
+This provides a component for the rebranded Quby thermostat as provided by
+Eneco.
+"""
+
+from homeassistant.components.climate import (ClimateDevice, ATTR_TEMPERATURE)
+from homeassistant.const import TEMP_CELSIUS
+import homeassistant.components.toon as toon_main
+
+STATE_COMFORT = 'Comfort'
+STATE_HOME = 'Home'
+STATE_AWAY = 'Away'
+STATE_SLEEP = 'Sleep'
+STATE_MANUAL = 'Manual'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup thermostat."""
+    # Add toon
+    add_devices((ThermostatDevice(hass), ), True)
+
+
+class ThermostatDevice(ClimateDevice):
+    """Interface class for the toon module and HA."""
+
+    def __init__(self, hass):
+        """Initialize the device."""
+        self._name = 'Toon van Eneco'
+        self.hass = hass
+        self.thermos = hass.data[toon_main.TOON_HANDLE]
+
+        # set up internal state vars
+        self._state = None
+        self._temperature = None
+        self._setpoint = None
+        self._operation_list = [STATE_COMFORT, STATE_HOME, STATE_AWAY,
+                                STATE_SLEEP, STATE_MANUAL]
+
+    @property
+    def name(self):
+        """Name of this Thermostat."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Polling is required."""
+        return True
+
+    @property
+    def temperature_unit(self):
+        """The unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_operation(self):
+        """Return current operation i.e. comfort, home, away."""
+        state = self.thermos.get_data('state')
+        return state

```

Comment:
That's not how Home Assistant works. You will need to implement the states as defined by the climate component.